conda_forge_tick/migrators/pip_check.py

```diff
@@ -0,0 +1,245 @@
+    if m:
+        tup = m.group(1, 2, 3, 4)
+        return mapping.get(tup, line)
+    else:
+        return line
+
+
+def _has_python_in_host(host):
+    host_set = {r.split(" ")[0] for r in host}
+    return bool(host_set & set(["python"]))
+
+
+def _gen_keys_selector(meta, base):
+    for key in meta.keys():
+        if key == base or key.startswith(base + '__conda-selector'):
+            yield key, meta[key]
+
+
+def _has_key_selector(meta, base):
+    return len([val for val in _gen_keys_selector(meta, base)]) > 0
+
+
+def _adjust_test_dict(meta, key, mapping, groups, parent_group=None):
+    if _has_key_selector(meta[key], 'requires'):
+        for _key, val in _gen_keys_selector(meta[key], 'requires'):
+            val.append('pip')
+            if _key in groups:
+                val.yaml_add_eol_comment(
+                    '# [%s]' % groups[_key][3],
+                    len(val) - 1,
+                    column=2,
+                )
+    elif _has_key_selector(meta[key], 'requirements'):
+        for _key, val in _gen_keys_selector(meta[key], 'requirements'):
+            val.append('pip')
+            if _key in groups:
+                val.yaml_add_eol_comment(
+                    '# [%s]' % groups[_key][3],
+                    len(val) - 1,
+                    column=80,
+                )
+    else:
+        new_seq = ruamel.yaml.comments.CommentedSeq()
+        new_seq.append('pip')
+        meta[key]['requires'] = new_seq
+        if parent_group is not None:
+            new_seq.yaml_add_eol_comment(
+                '# [%s]' % parent_group[3],
+                0,
+                column=80,
+            )
+            meta[key].yaml_add_eol_comment(
+                '# [%s]' % parent_group[3],
+                'requires',
+                column=80,
+            )
+
+    if _has_key_selector(meta[key], 'commands'):
+        for _key, val in _gen_keys_selector(meta[key], 'commands'):
+            val.append('{{ TEST_PYTHON }} -m pip check')
+            if _key in groups:
+                val.yaml_add_eol_comment(
+                    '# [%s]' % groups[_key][3],
+                    len(val) - 1,
+                    column=80,
+                )
+    else:
+        new_seq = ruamel.yaml.comments.CommentedSeq()
+        new_seq.append('{{ TEST_PYTHON }} -m pip check')
+        meta[key]['commands'] = new_seq
+        if parent_group is not None:
+            new_seq.yaml_add_eol_comment(
+                '# [%s]' % parent_group[3],
+                0,
+                column=80,
+            )
+            meta[key].yaml_add_eol_comment(
+                '# [%s]' % parent_group[3],
+                'commands',
+                column=80,
+            )
+
+
+class PipCheckMigrator(MiniMigrator):
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        """run pip check if we see python in any host sections"""
+        return "python" not in attrs['requirements'].get('host', set())
```

Comment:
```suggestion
        return "python" not in attrs['requirements'].get('host', set()) or attrs['requirements'].get('build', set()) or set()
```
This way we handle super old syntax where packages used build not host. (we should update that syntax but that might be out of scope for this PR)